darts/models/theta.py

```diff
@@ -115,14 +123,228 @@ def predict(self, n: int) -> 'TimeSeries':
+        """
+
+        super().__init__()
+
+        self.model = None
+        self.drift = None
+        self.coef = 1
+        self.mean = 1
+        self.length = 0
+        self.theta = theta
+        self.is_seasonal = False
+        self.seasonality = None
+        self.season_period = seasonality_period
+        self.model_mode = model_mode
+        self.season_mode = season_mode
+        self.trend_mode = trend_mode
+        self.wses = 0 if self.theta == 0 else (1 / theta)
+        self.wdrift = 1 - self.wses
+        self.fitted_values = None
+
+        raise_if_not(model_mode in Model,
+                     "Unknown value for model_mode: {}.".format(model_mode), logger)
+        raise_if_not(trend_mode in Trend,
+                     "Unknown value for trend_mode: {}.".format(trend_mode), logger)
+        raise_if_not(season_mode in Season,
+                     "Unknown value for season_mode: {}.".format(season_mode), logger)
+
+        # Remark on the values of the theta parameter:
+        # - if theta = 1, then the theta method restricts to a simple exponential smoothing (SES)
+        # - if theta = 0, then the theta method restricts to a simple `trend_mode` regression.
+
+    def fit(self, ts, component_index: Optional[int] = None):
+        super().fit(ts, component_index)
+
+        self.length = len(ts)
+        # normalization of data
+        self.mean = ts.mean().mean()
+        if self.mean == 0:
+            self.mean = 1.
+        new_ts = ts / self.mean
+
+        if (new_ts <= 0).values.any():
+            self.model_mode = Model.ADDITIVE
+            self.trend_mode = Trend.LINEAR
+            self.season_mode = Season.ADDITIVE
+            logger.warn("Time series has negative values. Fallback to additive and linear model")
+
+        # Check for statistical significance of user-defined season period
+        # or infers season_period from the TimeSeries itself.
+        if self.season_mode is Season.NONE:
+            self.season_period = 1
+        if self.season_period is None:
+            max_lag = len(ts) // 2
+            self.is_seasonal, self.season_period = check_seasonality(ts, self.season_period, max_lag=max_lag)
+            logger.info('Theta model inferred seasonality of training series: {}'.format(self.season_period))
+        else:
+            # force the user-defined seasonality to be considered as a true seasonal period.
+            self.is_seasonal = self.season_period > 1
+
+        # Store and remove seasonality effect if there is any.
+        if self.is_seasonal:
+            _, self.seasonality = extract_trend_and_seasonality(new_ts, self.season_period,
+                                                                model=self.season_mode.value)
+            new_ts = remove_seasonality(new_ts, self.season_period, model=self.season_mode.value)
+
+        # Drift part of the decomposition
+        if self.trend_mode is Trend.LINEAR:
+            linreg = new_ts.univariate_values()
```

Comment:
You could start extracting the univariate values earlier in the method, in order to do it only once, and fail early if it's not univariate.